source/adapters/level_zero/device.cpp

```diff
@@ -747,11 +747,13 @@ ur_result_t urDeviceGetInfo(
   }
 
   case UR_DEVICE_INFO_GLOBAL_MEM_FREE: {
-    if (getenv("ZES_ENABLE_SYSMAN") == nullptr) {
-      setErrorMessage("Set ZES_ENABLE_SYSMAN=1 to obtain free memory",
-                      UR_RESULT_ERROR_UNINITIALIZED,
+    bool SysManEnv = getenv_tobool("ZES_ENABLE_SYSMAN", false);
+    if ((Device->Platform->ZedeviceToZesDeviceMap.size() == 0) && !SysManEnv) {
+      setErrorMessage("SysMan support is unavailable on this system. Please "
```

Comment:
`setErrorMessage` is only meant to be used together with `UR_RESULT_ERROR_ADAPTER_SPECIFIC`. Simply use the logger to print an error message.

Reply:
ok, I will delete the set error message.

Reply:
Deleted and replaced with the logger.